### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by PierreBarba**
# hello reviewer we hope you are having a great day, any suggestion will be welcome, thank you very much for the support 😄 🌴

## Project requirements
* Install CanCanCan in your project.
* Add a role column to the users table. Remember to use a migration for this.
* A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
    * For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
* A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
    * For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

Captura de pantalla 2023-04-20 a la(s) 9 24 28

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
   validates :name, presence: true
-  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  # validates :postsCounter, comparison: { greater_than_or_equal_to: 0 }
 
   def recent_posts
     posts.order(created_at: :desc).limit(3)
   end
+
+  def admin?
+    role == 'admin'
```
Comment (afanimartin) [inline app/models/user.rb:18]: - 👏🏿 this is cool

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -27,6 +28,13 @@ def create
     end
   end
 
+  def destroy
```
Comment (afanimartin) [inline app/controllers/posts_controller.rb:33]: - This before your actions so as to initialize `cancan`
```ruby
  class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

**Review by afanimartin — CHANGES_REQUESTED** [review]
Hi @PierreBarba, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- `cancan` gem installed
- Readme is professional
- Ability added to models

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Find me on slack
### Martin Afani

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by arslanbisharat** [comment]
Hi @PierreBarba,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**